### ISSUE TITLE
WinAfl instrument.exe ERRORS

### ISSUE THREAD

**Issue by DaniDevelop** [googleprojectzero/winafl#193]
Hello again haha :), 
I try to create an insturment file for my program to test the fuzzer, 

My input command is 
`instrument.exe --mode=afl --input-image=Program.exe --output-image=Program.instr.exe --force-decompose --multithread --cookie-check-hook`

But when I try to run this command i get errors which look like this : 

`
[0705/204753:ERROR:dia_util.cc(64)] Failed to create Dia object.
[0705/204753:ERROR:dia_util.cc(65)]   NoRegCoCreate failed with: [hr=0x800700c1, msg=%1 is not a valid Win32 application.]
[0705/204753:ERROR:dia_util.cc(66)]   CreateInstance failed with: [hr=0x80040154, msg=Class not registered]
[0705/204753:ERROR:pe_relinker.cc(66)] Unable to decompose module: C:\Users\danie\Desktop\winafl\bin32\Program.exe
[0705/204753:ERROR:instrumenter_with_relinker.cc(124)] Failed to initialize relinker.
`

Can someone help me please :) 
thanks, mrxsys.

**Comment by ifratric** [comment]
Did you build the binary you are trying to instrument with /PROFILE linker flag. Also note that static instrumentation works only on 32-bit binaries (see https://github.com/googleprojectzero/winafl/blob/master/readme_syzygy.md)

**Comment by ditah22** [comment]
Hi, I've got similar issue when running the example test_static.exe:

`C:\Users\ditah\Desktop\winafl\bin32>instrument.exe --mode=afl --input-image=test_static.exe --output-image=test_static.instr.exe`

the output is:
`
[0502/153213:INFO:application_impl.h(46)] Syzygy Instrumenter Version 0.8.32.0 (190dbfe).
[0502/153213:INFO:application_impl.h(48)] Copyright (c) Google Inc. All rights reserved.
[0502/153213:INFO:pe_relinker_util.cc(336)] Input PDB not specified, searching for it.
[0502/153213:INFO:pe_relinker_util.cc(362)] Using default output PDB path: C:\Users\ditah\Desktop\winafl\bin32\test_static.instr.exe.pdb
[0502/153213:INFO:pe_relinker.cc(138)] Input module : C:\Users\ditah\Desktop\winafl\bin32\test_static.exe
[0502/153213:INFO:pe_relinker.cc(139)] Input PDB    : .\test_static.pdb
[0502/153213:INFO:pe_relinker.cc(140)] Output module: C:\Users\ditah\Desktop\winafl\bin32\test_static.instr.exe
[0502/153213:INFO:pe_relinker.cc(141)] Output PDB   : C:\Users\ditah\Desktop\winafl\bin32\test_static.instr.exe.pdb
[0502/153213:INFO:pe_relinker.cc(57)] Decomposing module: C:\Users\ditah\Desktop\winafl\bin32\test_static.exe
[0502/153213:ERROR:dia_util.cc(64)] Failed to create Dia object.
[0502/153213:ERROR:dia_util.cc(65)]   NoRegCoCreate failed with: [hr=0x8007007e, msg=The specified module could not be found.]
[0502/153213:ERROR:dia_util.cc(66)]   CreateInstance failed with: [hr=0x80040154, msg=Class not registered]
[0502/153213:ERROR:pe_relinker.cc(66)] Unable to decompose module: C:\Users\ditah\Desktop\winafl\bin32\test_static.exe
[0502/153213:ERROR:instrumenter_with_relinker.cc(124)] Failed to initialize relinker.
`

Thanks,
Dor

**Comment by ditah22** [comment]
Hi, I've got similar issue when running the example test_static.exe:

`C:\Users\ditah\Desktop\winafl\bin32>instrument.exe --mode=afl --input-image=test_static.exe --output-image=test_static.instr.exe --force-decompose`

`[0502/153213:INFO:application_impl.h(46)] Syzygy Instrumenter Version 0.8.32.0 (190dbfe).`
`[0502/153213:INFO:application_impl.h(48)] Copyright (c) Google Inc. All rights reserved.`
`[0502/153213:INFO:pe_relinker_util.cc(336)] Input PDB not specified, searching for it.`
`[0502/153213:INFO:pe_relinker_util.cc(362)] Using default output PDB path: C:\Users\ditah\Desktop\winafl\bin32\test_static.instr.exe.pdb`
`[0502/153213:INFO:pe_relinker.cc(138)] Input module : C:\Users\ditah\Desktop\winafl\bin32\test_static.exe`
`[0502/153213:INFO:pe_relinker.cc(139)] Input PDB    : .\test_static.pdb`
`[0502/153213:INFO:pe_relinker.cc(140)] Output module: C:\Users\ditah\Desktop\winafl\bin32\test_static.instr.exe`
`[0502/153213:INFO:pe_relinker.cc(141)] Output PDB   : C:\Users\ditah\Desktop\winafl\bin32\test_static.instr.exe.pdb`
`[0502/153213:INFO:pe_relinker.cc(57)] Decomposing module: C:\Users\ditah\Desktop\winafl\bin32\test_static.exe`
`[0502/153213:ERROR:dia_util.cc(64)] Failed to create Dia object.`
`[0502/153213:ERROR:dia_util.cc(65)]   NoRegCoCreate failed with: [hr=0x8007007e, msg=The specified module could not be found.]`
`[0502/153213:ERROR:dia_util.cc(66)]   CreateInstance failed with: [hr=0x80040154, msg=Class not registered]`
`[0502/153213:ERROR:pe_relinker.cc(66)] Unable to decompose module: C:\Users\ditah\Desktop\winafl\bin32\test_static.exe`
`[0502/153213:ERROR:instrumenter_with_relinker.cc(124)] Failed to initialize relinker.`

Thanks,
Dor

**Comment by nevilad** [comment]
It can't open `IDiaDataSource` interface to get access to debugging symbols. Have you any `msdiaXXX.dll` registered?

**Comment by ditah22** [comment]
I tried to follow this [readme_syzygy](https://github.com/googleprojectzero/winafl/blob/master/readme_syzygy.md), I ran `regsvr32 /s msdia140.dll` in cmd before `instrument.exe --mode=afl --input-image=test_static.exe --output-image=test_static.instr.exe --force-decompose` if that's what you meant..

Do I need to something else?

**Comment by nevilad** [comment]
Did `regsvr32 /s msdia140.dll` ran successfully?

**Comment by ditah22** [comment]
Didn't print any error...

![image](https://user-images.githubusercontent.com/80756400/117152806-0a6bd880-adc3-11eb-99e6-209adc518983.png)

**Comment by nevilad** [comment]
Looks like successfull registered msdia, but still can't load it. Try to run instrument.exe with your command line in dependency viewer. This will log al the errors in dll loading.

**Comment by ditah22** [comment]
I'm not sure if that's what you meant:
![image](https://user-images.githubusercontent.com/80756400/117569538-d3513c00-b0ce-11eb-8be9-2b65b69ec0f3.png)
And I'm not sure what to do with those missing dlls

**Comment by nevilad** [comment]
No, that's static view. I meant a profiling run, `Profile->Start profiling run...` menu in the original dependency walker.

**Comment by kilimnik** [comment]
I had the same Issue and found a fix for it. First of all I noticed when executing the `regsvr32 /s msdia140.dll` command, the `/s` is for silent. That's why it didn't error.
To register it successfully I opened an elevated command prompt went to this `C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\DIA SDK\bin>` path and executed `regsvr32 msdia140.dll`. This resulted in a Message Box with a success message.

After that the instrument command was successful too.
